src/SecTester.Scan/DefaultScanFactory.cs

```diff
@@ -0,0 +1,111 @@
+  {
+    _scans = scans ?? throw new ArgumentNullException(nameof(scans));
+    _configuration = configuration ?? throw new ArgumentNullException(nameof(configuration));
+    _systemTimeProvider = systemTimeProvider ?? throw new ArgumentNullException(nameof(systemTimeProvider));
+    _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+  }
+
+  public async Task<Scan> CreateScan(ScanSettingsOptions settingsOptions, ScanOptions? options)
+  {
+    var scanConfig = await BuildScanConfig(new ScanSettings(settingsOptions)).ConfigureAwait(false);
+    var scanId = await _scans.CreateScan(scanConfig).ConfigureAwait(false);
+
+    return new Scan(scanId, _scans, _logger, options ?? new ScanOptions());
+  }
+
+  private async Task<ScanConfig> BuildScanConfig(ScanSettings scanSettings)
+  {
+    var fileId = await CreateAndUploadHar((Target.Target)scanSettings.Target).ConfigureAwait(false);
+
+    return new ScanConfig(scanSettings.Name!)
+    {
+      FileId = fileId,
+      Smart = scanSettings.Smart,
+      PoolSize = scanSettings.PoolSize,
+      SkipStaticParams = scanSettings.SkipStaticParams,
+      Module = Module.Dast,
+      DiscoveryTypes = DefaultDiscoveryTypes,
+      AttackParamLocations = scanSettings.AttackParamLocations,
+      Tests = scanSettings.Tests,
+      Repeaters = scanSettings.RepeaterId is null ? default : new List<string> { scanSettings.RepeaterId },
+      SlowEpTimeout =
+        scanSettings.SlowEpTimeout is null ? default : (int)scanSettings.SlowEpTimeout.Value.TotalSeconds,
+      TargetTimeout =
+        scanSettings.TargetTimeout is null ? default : (int)scanSettings.TargetTimeout.Value.TotalSeconds
+    };
+  }
+
+  private async Task<string> CreateAndUploadHar(Target.Target target)
+  {
+    var filename = GenerateFileName(target.Url);
+    var har = await CreateHar(target).ConfigureAwait(false);
+
+    return await _scans.UploadHar(new UploadHarOptions(har, filename, true)).ConfigureAwait(false);
+  }
+
+  private static string GenerateFileName(string url)
+  {
+    var host = new Uri(url).Host;
+
+    host = host.Length <= MaxSlugLength ? host : host.Substring(0, MaxSlugLength);
+
+    return $"{host.TrimEnd(".-".ToCharArray())}-{Guid.NewGuid()}.har";
+  }
+
+  private async Task<Entry> CreateHarEntry(Target.Target target)
+  {
+    return new Entry(_systemTimeProvider.Now)
+    {
+      Time = 0,
+      Request = await target.ToHarRequest().ConfigureAwait(false),
+      Timings = new Timings { Send = 0, Receive = 0, Wait = 0 },
+      Cache = new Cache(),
+      Response = new Response(new Content { Size = -1, MimeType = "text/plain" })
+      {
+        HttpVersion = "HTTP/1.1",
+        Status = 200,
+        StatusText = "OK",
+        HeadersSize = -1,
+        BodySize = -1,
+        RedirectUrl = "",
+        Cookies = new List<Cookie>(),
+        Headers = new List<Header>()
```

Comment:
All of them have default values

```suggestion
        RedirectUrl = ""
```